### PR TITLE
KAFKA-16004: auto-commit inflight improved logs, docs and tests

### PR DESCRIPTION
**Pull request by lianetm**
Minor changes for improving the logging and docs related to the auto-commit inflight logic, also adding tests to ensure the expected behaviour:

- auto-commit on the interval does not send a request if another one inflight, and it sends the next as soon as a response is received (without waiting for the full interval again)
- auto-commit before revocation always send a request (even if another one from auto-commit on interval is in-flight), to ensure the latest is committed before revoking partitions. 

No changes in logic, just adding tests, docs and minor refactoring.

### CODE REVIEW

**Comment by lianetm** [comment]
Hey @lucasbru, could you take a look at this when you have a chance? Thanks!

**Comment by lianetm** [comment]
Hey @cadonna, could you take a look if you have some time too? Thanks!

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java [hunk 1]
```diff
@@ -498,6 +538,40 @@ public void testAutoCommitEmptyDoesNotLeaveInflightRequestFlagOn() {
         verify(commitRequestManger, times(2)).resetAutoCommitTimer();
     }
 
+    @Test
+    public void testAutoCommitOnIntervalSkippedIfPreviousOneInFlight() {
+        TopicPartition t1p = new TopicPartition("topic1", 0);
+        subscriptionState.assignFromUser(singleton(t1p));
+        subscriptionState.seek(t1p, 100);
+
+        CommitRequestManager commitRequestManger = create(true, 100);
```
Comment (cadonna) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java]: nit:
```suggestion
        CommitRequestManager commitRequestManager = create(true, 100);
```

**Review by cadonna — APPROVED** [review]
Thanks for the PR @lianetm !

LGTM!

I just found one typo.

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java [hunk 2]
```diff
@@ -498,6 +538,40 @@ public void testAutoCommitEmptyDoesNotLeaveInflightRequestFlagOn() {
...
+        commitRequestManger.maybeAutoCommitAsync();
+        assertPoll(0, commitRequestManger);
+        verify(commitRequestManger, never()).resetAutoCommitTimer();
+
+        // When a response for the inflight is received, a next auto-commit should be sent when
+        // polling the manager.
+        inflightCommitResult.onComplete(
+            mockOffsetCommitResponse(t1p.topic(), t1p.partition(), (short) 1, Errors.NONE));
+        assertPoll(1, commitRequestManger);
+    }
```
Comment (cadonna) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java]:
```suggestion
        // Send auto-commit request that will remain in-flight without a response
        time.sleep(100);
        commitRequestManager.updateAutoCommitTimer(time.milliseconds());
        commitRequestManager.maybeAutoCommitAsync();
        List<NetworkClientDelegate.FutureCompletionHandler> futures = assertPoll(1, commitRequestManager);
        assertEquals(1, futures.size());
        NetworkClientDelegate.FutureCompletionHandler inflightCommitResult = futures.get(0);
        verify(commitRequestManager, times(1)).resetAutoCommitTimer();
        clearInvocations(commitRequestManager);

        // After next interval expires, no new auto-commit request should be sent. The interval
        // should not be reset either, to ensure that the next auto-commit is sent out as soon as
        // the inflight receives a response.
        time.sleep(100);
        commitRequestManager.updateAutoCommitTimer(time.milliseconds());
        commitRequestManager.maybeAutoCommitAsync();
        assertPoll(0, commitRequestManager);
        verify(commitRequestManager, never()).resetAutoCommitTimer();

        // When a response for the inflight is received, a next auto-commit should be sent when
        // polling the manager.
        inflightCommitResult.onComplete(
            mockOffsetCommitResponse(t1p.topic(), t1p.partition(), (short) 1, Errors.NONE));
        assertPoll(1, commitRequestManager);
    }
```

**Comment by cadonna** [comment]
@lianetm I found the same typo throughout the test class. Do you want to fix it in the entire file?

**Comment by lianetm** [comment]
Thanks @cadonna! fixed it in the entire file.

**Pull request merged**